src/pubsub-message.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-env mocha */
+'use strict'
+
+const expect = require('chai').expect
+const isNode = require('detect-node')
+const PubsubMessage = require('../../src/pubsub-message') // eslint-disable-line no-unused-vars
+const PubsubMessageUtils = require('../../src/pubsub-message-utils')
+
+const topicName = 'js-ipfs-api-tests'
+
+// NOTE!
+// These tests are skipped for now until we figure out the
+// final data types for the messages coming over the wire
+
+describe('.pubsub-message', () => {
+  if (!isNode) {
+    return
+  }
+
+  it.skip('create message', () => {
+    // TODO
+  })
+
+  it.skip('deserialize message from JSON object', () => {
+    const obj = {
+      from: 'BI:ۛv�m�uyѱ����tU�+��#���V',
+      data: 'aGk=',
+      seqno: 'FIlj2BpyEgI=',
+      topicIDs: [ topicName ]
+    }
+    try {
+      const message = PubsubMessageUtils.deserialize(obj)
+      expect(message.from).to.equal('AAA')
+      expect(message.data).to.equal('hi')
+      expect(message.seqno).to.equal('\u0014�c�\u001ar\u0012\u0002')
+      expect(message.topicIDs.length).to.equal(1)
+      expect(message.topicIDs[0]).to.equal(topicName)
+    } catch (e) {
+      expect(e).to.not.exist
+    }
+  })
+
+  describe('immutable properties', () => {
+    const message = PubsubMessageUtils.create('A', 'hello', '123', ['hello world'])
+
+    it('from', () => {
```

Comment:
Not sure if that's the right fix. You just added a expect in the try part, which would not get run if there is an error and now we're doing different assertions depending on if the code throws an exception or not. The tests should not be able to change the executed code path.